tests/validation_tests/test_tf_r4_specific_validations.py

```diff
@@ -1,14 +1,82 @@
+import os
+
 import pandas as pd
+import pytest
+from openpyxl import Workbook
+from openpyxl.drawing.image import Image
 
 from core.validation.specific_validations.towns_fund_round_four import (
     TownsFundRoundFourValidationFailure,
     validate,
     validate_programme_risks,
     validate_project_risks,
+    validate_sign_off,
 )
 
 
-def test_validate_returns_failures(mocker):
+@pytest.fixture
+def excel_file_success(tmp_path):
+    excel_file = tmp_path / "fake_excel.xlsx"
+
+    wb = Workbook()
+    sheet = wb.active
+    sheet.title = "8 - Review & Sign-Off"
+
+    current_dir = os.getcwd()
+    image_path = os.path.join(current_dir, "tests/validation_tests/resources", "fake_signature.png")
+    img1 = Image(image_path)
+    img2 = Image(image_path)
+    sheet.add_image(img1, "C10")
+    sheet.add_image(img2, "C17")
+    sheet["B8"] = "Field"
+    sheet["C8"] = "Value"
+    sheet["B9"] = "Field"
+    sheet["C9"] = "Value"
+    sheet["B11"] = "Field"
+    sheet["C11"] = "Value"
+    sheet["B15"] = "Field"
+    sheet["C15"] = "Value"
+    sheet["B16"] = "Field"
+    sheet["C16"] = "Value"
+    sheet["B18"] = "Field"
+    sheet["C18"] = "Value"
+
+    wb.save(excel_file)
+
+    return excel_file
+
+
+@pytest.fixture
+def excel_file_failure(tmp_path):
+    excel_file = tmp_path / "fake_excel.xlsx"
+
+    wb = Workbook()
+    sheet = wb.active
+    sheet.title = "8 - Review & Sign-Off"
+
+    current_dir = os.getcwd()
+    image_path = os.path.join(current_dir, "tests/validation_tests/resources", "fake_signature.png")
+    img1 = Image(image_path)
+    sheet.add_image(img1, "C10")
+    sheet["C17"] = None
+    sheet["B8"] = "Name"
+    sheet["B9"] = "Field Value"
+    sheet["C9"] = "Field Value"
+    sheet["B11"] = "Field Value"
+    sheet["C11"] = "Field Value"
+    sheet["B15"] = "Field Value"
+    sheet["C15"] = "Field Value"
+    sheet["B16"] = "Field Value"
+    sheet["C16"] = "Field Value"
+    sheet["B18"] = "Field Value"
+    sheet["C18"] = "Field Value"
+
+    wb.save(excel_file)
+
+    return excel_file
+
+
+def test_validate_returns_failures(mocker, excel_file_success):
```

Comment:
Instead of passing a success excel file to these two tests (`test_validate_returns_failures` & `test_validate_returns_empty_list`), can we just mock the sign off validation function to return `None` instead. That way we're isolating just the `validate` function's logic.

Reply:
We can, but we still have to pass a second parameter to validate.
```
failures = validate(mock_workbook, None)
```
works if we mock the sign off validation function to return None.